### ISSUE TITLE
Disconnection after death

### ISSUE THREAD

**Issue by Viskus** [ZhengPeiRu21/mod-challenge-modes#28]
Hi ZhengPeiRu21,
Is it an intended feature to disconnect after you click release spirit or try and revive your character?
Maybe I'm missing some config?

This is on HardCore Mode (I'm not working with any other mode currently)

**Comment by netweaver** [comment]
I suggest that in hardcore mode, characters should be allowed to log in after death, but cannot be resurrected. It is best to maintain a soul state in order to handle guild and other matters

**Comment by ZhengPeiRu21** [comment]
Yes, this is intended behavior. Currently AzerothCore does not have the necessary hooks to prevent resurrection. If these can be added to the core, then the module can be revised to allow logging in as a ghost.

**Comment by Viskus** [comment]
I agree that the character should not be deleted and should be able to login etc. I will put up a request in AC repo to see if anyone else would like to see the hooks added to the core, thanks again

**Comment by alexpec1** [comment]
为什么我用硬核模式死亡后，退出游戏重新登录后再点复活，人物就可以正常复活了呢？不是应该是不能复活的吗？我这个是什么原因导致呢，用的也是作者的这个模块

**Comment by yuanf225** [comment]
> I agree that the character should not be deleted and should be able to login etc. I will put up a request in AC repo to see if anyone else would like to see the hooks added to the core, thanks again

No PR was found for this hook. Did you submit this request?

**Comment by netweaver** [comment]
@yuanf225 i will  PR soon ( both on AC and eluna side), tested ok by creating a lua script fully implement hardcore mode.

**Comment by Viskus** [comment]
I've added one here for AZ core - https://github.com/azerothcore/azerothcore-wotlk/issues/17370

**Comment by netweaver** [comment]
pull request https://github.com/azerothcore/azerothcore-wotlk/pull/17460
